src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -0,0 +1,358 @@
+                                     .WithChangedOption(KeybindingResetOptions.NeedsReset, true);
+                    needsReset = true;
+                    _workspace.Options = options;
+                }
+                else
+                {
+                    needsReset = options.GetOption(KeybindingResetOptions.NeedsReset);
+                }
+
+                if (needsReset)
+                {
+                    ShowGoldBar();
+                }
+            }
+        }
+
+        private bool IsReSharperEnabled()
+        {
+            AssertIsForeground();
+
+            if (_oleCommandTarget == null)
+            {
+                var oleServiceProvider = (IOleServiceProvider)_serviceProvider.GetService(typeof(IOleServiceProvider));
+                _oleCommandTarget = (IOleCommandTarget)oleServiceProvider.QueryService(VSConstants.SID_SUIHostCommandDispatcher);
+            }
+
+            var cmds = new OLECMD[1];
+            cmds[0].cmdID = SuspendId;
+            cmds[0].cmdf = 0;
+
+            ErrorHandler.ThrowOnFailure(_oleCommandTarget.QueryStatus(ReSharperCommandGroup, (uint)cmds.Length, cmds, IntPtr.Zero));
+
+            // When ReSharper is enabled, the ReSharper_Suspend command has the Enabled | Supported flags. When disabled, it has Invisible | Supported.
+            return ((OLECMDF)cmds[0].cmdf).HasFlag(OLECMDF.OLECMDF_ENABLED);
+        }
+
+        private void ShowGoldBar()
+        {
+            AssertIsForeground();
+
+            // If the gold bar is already open, do not show
+            if (_infoBarOpen)
+            {
+                return;
+            }
+
+            _infoBarOpen = true;
+
+            string message;
+            if (_experimentationService.IsExperimentEnabled(InternalFlightName))
+            {
+                message = ServicesVSResources.We_noticed_you_suspended_ReSharper_Ultimate_Restore_Visual_Studio_keybindings_to_continue_to_navigate_and_refactor;
+            }
+            else if (_experimentationService.IsExperimentEnabled(ExternalFlightName))
+            {
+                message = ServicesVSResources.We_noticed_your_keybindings_are_broken;
+            }
+            else
+            {
+                // Should never have gotten to checking this if one of the flights isn't enabled.
+                throw ExceptionUtilities.Unreachable;
+            }
+
+            KeybindingsResetLogger.Log("InfoBarShown");
+            var infoBarService = _workspace.Services.GetRequiredService<IInfoBarService>();
+            infoBarService.ShowInfoBarInGlobalView(
+                message,
+                new InfoBarUI(title: ServicesVSResources.Restore_Visual_Studio_keybindings,
+                              kind: InfoBarUI.UIKind.Button,
+                              action: RestoreVsKeybindings),
+                new InfoBarUI(title: ServicesVSResources.Use_Keybindings_for_extensions,
+                              kind: InfoBarUI.UIKind.Button,
```

Comment:
shouldn't this be hyperlink and the next one a button?

Reply:
This is what @kuhlenh asked for.